src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -462,6 +465,75 @@ s.searchParameterInfo.Name switch
             }
         }
 
+        private static void RemoveRedundantParameters(IndentedStringBuilder stringBuilder, SqlParameterCollection sqlParameterCollection)
+        {
+            var commandText = stringBuilder.ToString();
+            if (!commandText.Contains("cte", StringComparison.OrdinalIgnoreCase))
+            {
+                commandText = commandText.Replace("DISTINCT ", string.Empty, StringComparison.OrdinalIgnoreCase);
+                if (!commandText.Contains(" OR ", StringComparison.OrdinalIgnoreCase))
+                {
+                    commandText = RemoveRedundantComparisons(commandText, sqlParameterCollection, '>');
+                    commandText = RemoveRedundantComparisons(commandText, sqlParameterCollection, '<');
+                }
+            }
+
+            stringBuilder.Clear();
```

Comment:
As I see, the intention here is to remove DISTINCTs and ORs. 
Does it make sense to keep 'stringBuilder' as it is, and do not changed it? 

Are the changes in lines 481 and 482 required if there are 'ctes'?

Reply:
No, that is a good point. If their are CTEs we can just return right away.